### ISSUE TITLE
bazel-bot failing with wrong bazel version

### ISSUE THREAD

**Issue by chingor13** [googleapis/repo-automation-bots#4058]
```
++ bazel query --google_default_credentials --remote_cache=https://storage.googleapis.com/repo-automation-bots-bazel-cache 'filter("-(go|csharp|java|php|ruby|nodejs|py)$", kind("rule", //...:*))'
++ grep -v -E ':(proto|grpc|gapic)-.*-java$'
Extracting Bazel installation...
Starting local Bazel server and connecting to it...
INFO: Invocation ID: 4086668b-6920-4131-8350-d5f5fce503da
Loading: 0 packages loaded
Loading: 0 packages loaded
Loading: 0 packages loaded
Loading: 0 packages loaded
ERROR: Traceback (most recent call last):
	File "/workspace/googleapis/WORKSPACE", line 93, column 22, in <toplevel>
		go_rules_dependencies()
	File "/builder/home/.cache/bazel/_bazel_root/7566ced0eec373338a04eb931067c421/external/io_bazel_rules_go/go/private/repositories.bzl", line 39, column 23, in go_rules_dependencies
		versions.check(MINIMUM_BAZEL_VERSION, bazel_version = native.bazel_version)
	File "/builder/home/.cache/bazel/_bazel_root/7566ced0eec373338a04eb931067c421/external/io_bazel_rules_go/go/private/skylib/lib/versions.bzl", line 104, column 13, in _check_bazel_version
		fail("\nCurrent Bazel version is {}, expected at least {}\n".format(
Error in fail: 
Current Bazel version is 4.0.0, expected at least 4.2.1
ERROR: error loading package '': Encountered error while reading extension file 'repositories.bzl': no such package '@com_googleapis_gapic_generator_go//': error loading package 'external': Could not load //external package
Loading: 0 packages loaded
Loading: 0 packages loaded
+ targets=
```

**Comment by chingor13** [comment]
We are rebuilding the our bazel-bot image from the base `gcr.io/gapic-images/googleapis:prod` image.

**Comment by chingor13** [comment]
> Where does this restriction come from? I don't see it in https://github.com/googleapis/googleapis/blob/master/WORKSPACE

Looks like it's an imported rule: https://github.com/googleapis/googleapis/blob/21ef8485aa4129268baa07c686fd6b60ba8f6f22/WORKSPACE#L93

**Comment by chingor13** [comment]
Image rebuild did not help

**Comment by chingor13** [comment]
The bazelisk docker image build is running currently, but will like take a long time as the commit probably touches every API

**Comment by chingor13** [comment]
bazel-bot is now working again with the built image from #4059.

Downgrading this to a p1 until we can resolve the tests and get that merged.

**Comment by chingor13** [comment]
Fixed in #4059
